### PR TITLE
Add motes and `NockStack` allocation safety

### PR DESCRIPTION
**Pull request by ashelkovnykov**
#182 has been split into 2 PRs: this one and #202. This one adds error motes and adds `NockStack` safety checks by returning the null pointer when allocating on the `NockStack` runs out of memory.

### CODE REVIEW

**Comment by matthew-levan** [comment]
Review is in-progress. Meanwhile, I've attempted to boot a fake `zod` and got this:
```
gall: installing %dojo
gall: installing %eth-watcher
gall: installing %hood
drum: link [~zod %dojo]
kiln: boot
gall: installing %herm
gall: installing %lens
gall: installing %ping
gall: installing %spider
pier: (9): play: done
---------------- playback complete ----------------
vere: checking version compatibility
lick init mkdir zod/.urb/dev
ames: live on 31337 (localhost only)
conn: listening on zod/.urb/conn.sock
lick: %born failure;
http: web ipier: work exit: status 0, signal 6081
http: loopback live on http://localhost:12322
pier (20): live
pier: EOF 
```

**Comment by matthew-levan** [comment]
This all looks good. I'm going to try to track down why I can't boot a fake ship with it today, but I have a hunch that your work in `mem.rs`, in particular, may be what I'm missing in my guard page branch for raising `SIGSEGV` signals everywhere we should be (i.e, when we try to allocate across stack/alloc pointers, inwards or outwards).

**Comment by matthew-levan** [comment]
Ok I was able to boot a fake `zod` after increasing `PMA_GROW_SIZE_p` to `10000000`.

**Review by matthew-levan — APPROVED** [review]

**Comment by eamsden** [comment]
@ashelkovnykov please resolve merge conflicts before further review

**Comment by ashelkovnykov** [comment]
> Ok I was able to boot a fake `zod` after increasing `PMA_GROW_SIZE_p` to `10000000`.
@matthew-levan You're correct, though I'm seeing the same behaviour on `status`.

File: rust/ares/src/mem.rs
Comment (matthew-levan) [inline rust/ares/src/mem.rs:1]: On further testing, I think these changes need to be reviewed more closely.

**Review by matthew-levan — DISMISSED** [review]

File: rust/ares/src/mem.rs
Comment (ashelkovnykov) [reply rust/ares/src/mem.rs:1]: Can you elaborate?

Comment (ashelkovnykov) [reply rust/ares/src/mem.rs:1]: @matthew-levan Upon further review: you're correct. The `null_ptr` produced by `NockStack` allocations in this PR only made sense in the original PR. With this PR, they cause an unhandled `SIGSEGV` on OOM (which is still better than silently clobbering the `NockStack`). However, these changes need to be undone in #202 (assuming that this PR is merged first).

**Review by matthew-levan — APPROVED** [review]

**Pull request merged**